src/pages/index.js

```diff
@@ -64,55 +80,176 @@ const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
 // Close image popup preview
 cardPreviewPopup.close();
 
+/* -------------------------------------------------------------------------- */
+/*                                Delete Popup                                */
+/* -------------------------------------------------------------------------- */
+
+// Create the delete popup instance
+const deletePopup = new Popup({ popupSelector: selectors.deletePopup });
+
+deletePopup.setEventListeners();
+
 /* -------------------------------------------------------------------------- */
 /*                                Card Section                                */
 /* -------------------------------------------------------------------------- */
 
 // This function creates a new card
-function createCard(data) {
-    const cardElement = new Card({ data, handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
-
-    }}, selectors.cardTemplate);
+function createCard(data, userId) {
+    const cardElement = new Card({ 
+        data,
+        handleImageClick: (imageData) => {
 
+            // Open image popup on click
+            cardPreviewPopup.open(imageData);
+        },
+        handleDeleteClick: () => {
+
+            // Open confirmation popup on click
+            deletePopup.open();
+
```

Comment:
Note, that it's needed to remove the card only after confirmation. You should implement `PopupWithConfirmation` class, which will handle the confirmation functionality.

Reply:
Thank you for the example. I've added the PopupWithConfirmation class now.